sendbuffer.go

```diff
@@ -43,24 +42,30 @@ func newSendBuffer(defaultHeader []byte, w io.Writer, windowSize int) *sendBuffe
 		writeTimer:     time.NewTimer(largeTimeout),
 	}
 	buf.closed.Add(1)
-	ops.Go(func() { buf.sendLoop(w) })
+	ops.Go(func() { buf.sendLoop(out) })
 	return buf
 }
 
-func (buf *sendBuffer) sendLoop(w io.Writer) {
+func (buf *sendBuffer) sendLoop(out chan []byte) {
+	closeTimedOut := make(chan interface{})
+
 	sendRST := false
 	defer func() {
 		if sendRST {
 			// Send an RST frame with the streamID
-			w.Write(withFrameType(buf.defaultHeader, frameTypeRST))
+			select {
+			case out <- withFrameType(buf.defaultHeader, frameTypeRST):
+				// okay
+			case <-closeTimedOut:
```

Comment:
I don't think we had a problem here, but it's better to be safe than sorry and not block on writing if we've closed.